src/common/transformations/src/transformations/op_conversions/convert_avgpool_downgrade.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/convert_avgpool_downgrade.hpp"
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/avg_pool.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::pass::ConvertAvgPool14ToAvgPool1::ConvertAvgPool14ToAvgPool1() {
+    MATCHER_SCOPE(ConvertAvgPool14ToAvgPool1);
+
+    const auto avg_pool_v14_pattern = pattern::wrap_type<ov::op::v14::AvgPool>();
+
+    const matcher_pass_callback callback = [](pattern::Matcher& m) {
+        const auto avg_pool_v14 = std::dynamic_pointer_cast<ov::op::v14::AvgPool>(m.get_match_root());
+        if (!avg_pool_v14) {
+            return false;
+        }
+
+        const auto rounding_type_v14 = avg_pool_v14->get_rounding_type();
+        const auto rounding_type_v1 =
```

Comment:
Done for both `AvgPool` and `MaxPool`. I'm not sure if the subgraph should be placed in `utils` since the API for them is different, which would need a few workarounds for the constructors and runtime info copying. Please let me know if you think it's a good candidate for moving into utils.